dagstore_control.go

```diff
@@ -74,35 +76,39 @@ func (d *DAGStore) control() {
 
 			// trigger queued acquisition waiters.
 			for _, w := range s.wAcquire {
+				// optimistically increment the refcount to acquire the shard. The go-routine will send an `OpShardRelease` message
+				// to the event loop if it fails to acquire the shard.
 				s.refs++
 				go d.acquireAsync(tsk.ctx, w, s, s.mount)
 			}
 			s.wAcquire = s.wAcquire[:0]
 
 		case OpShardAcquire:
 			w := &waiter{ctx: tsk.ctx, outCh: tsk.outCh}
-			if s.state != ShardStateAvailable && s.state != ShardStateServing {
+			// TODO What if Shard is in the ShardStateErrored state  here ? For now, that will block the acquirer as
```

Comment:
@raulk The failure management/recovery mechanism will have to handle these cases.

Reply:
The acquire should fail immediately, returning an error in the response channel that wraps the shard's error.

Reply:
Taken care of in #51.